iron-overlay-behavior.html

```diff
@@ -392,10 +393,11 @@
       // Move the focus to the child node with [autofocus].
       this._applyFocus();
 
-      // Safari will apply the focus to the autofocus element when displayed for the first time,
-      // so we blur it. Later, _applyFocus will set the focus if necessary.
-      if (this.noAutoFocus && document.activeElement === this._focusNode) {
+      // Safari will apply the focus to the autofocus element when displayed
+      // for the first time, so we make sure to return the focus where it was.
+      if (this.noAutoFocus) {
         this._focusNode.blur();
+        this.__restoreFocusNode.focus();
```

Comment:
This is because of what Safari does when an `autofocus` element is displayed for the first time (see comment at 396). Not only we want to blur the `_focusNode`, but also move the focus to where it was before